src/bin/casr-cluster.rs

```diff
@@ -416,10 +416,69 @@ fn dedup(indir: &Path, outdir: Option<PathBuf>) -> error::Result<(usize, usize)>
     Ok((before, after))
 }
 
+/// Merge sub directory to main
+///
+/// # Arguments
+///
+/// * `sdir` - path to sub directory
+///
+/// * `mdir` - path to main directory
+///
+/// # Return value
+///
+/// Number of merged reports
+fn merge_dirs(sdir: &Path, mdir: &Path) -> error::Result<u64> {
+    let dir = fs::read_dir(mdir).with_context(|| {
+        format!(
+            "Error occurred while opening directory with Casr reports. File: {}",
+            mdir.display()
+        )
+    })?;
+
+    let mut mainhash = HashSet::new();
+    for entry in dir.flatten() {
+        if entry.path().extension().is_some() && entry.path().extension().unwrap() == "casrep" {
+            if let Ok(trace) = stacktrace(entry.path().as_path()) {
+                mainhash.insert(trace);
+            } else {
+                // return err?
```

Comment:
We may return error here and do not perform merge if output directory is corrupted.